utils/mil_ros_tools/online_bagger/nodes/online_bagger.py

```diff
@@ -0,0 +1,188 @@
+
+for example:
+
+    bagging_server = rospy.ServiceProxy('/online_bagger', SetBool)
+    bag_status = bagging_server(True)
+'''
+
+
+class OnlineBagger(object):
+    def __init__(self):
+        '''
+        - makes dictionary of dequeues
+        - subscribes to set of topics defined by the yaml file in directory
+        - have a streaming option that can be turned on and off in the callback
+        - option to bag data
+        - add a re subscribing option for anything that failed? maybe...?
+        '''
+
+        self.n = 0  # number of iterations
+        self.streaming = True
+        self.dumping = False
+        # self.resubscribe
+
+        self.get_params()
+        self.make_dicts()
+        self.subscribe()
+        rospy.loginfo('subscriber success: %s', self.subscriber_list)
+
+
+        self.bagging_service = rospy.Service('/online_bagger/dump', SetBool, self.start_bagging)
+
+    def get_params(self):
+
+        self.topics = rospy.get_param('/online_bagger/message_topics')
+        self.ram_limit = rospy.get_param('/online_bagger/ram_limit')*1.073e9 
+        self.stream_time = rospy.get_param('/online_bagger/stream_time')
+
+        rospy.loginfo('stream_time: %s seconds', self.stream_time)
+        rospy.loginfo('ram_limit: %s gb', self.ram_limit)
+
+    # def set_ram_limit(self)
+
+    #     # http://stackoverflow.com/questions/30269238/limit-memory-usage
+    #     # _, hard = resource.getrlimit(resource.RLIMIT_DATA)
+    #     resource.setrlimit(resource.RLIMIT_DATA, (self.ram_limit, hard))
+
+
+    def make_dicts(self):
+        '''
+        make dictionaries with deques() that will be filled with topics
+        with all of the 5topics defined by the yaml
+        '''
+
+        self.topic_list = {}
+        self.subscriber_list = []
+        for topic in self.topics:
+            self.topic_list[topic] = deque()
+            self.subscriber_list.append([topic, False])
+
+        rospy.loginfo('failed topics: %s', self.subscriber_list)
+
+    def subscribe(self):
+        '''
+        Immediately initiated with instance of the BagStream class:
+        subscribes to the set of topics defined in the yaml configuration file
+
+        Function checks subscription status True/False of each topic
+        if True: topic has already been sucessfully subscribed to
+        if False: topic still needs to be subscribed to and 
+        subscriber will be run
+
+        Each element in self.subscriber list is a list [topic, Bool]
+        where the Bool tells the current status of the subscriber (sucess/failure)
+        '''
+
+        for topic in self.subscriber_list:
+            if topic[1] is True:
+                pass
+            else:
+                msg_class = rostopic.get_topic_class(topic[0])
+                if msg_class[1] is None:
+                    pass
+                else:
+                    rospy.Subscriber(topic[0], msg_class[0], lambda msg, _topic=topic[0]: self.callback(msg, _topic))
```

Comment:
Same thing here:

                if msg_class[1] is not None:
                    rospy.Subscriber(topic[0], msg_class[0], lambda msg, _topic=topic[0]: self.callback(msg, _topic))
                    ...